src/integration/blockchain/monero/monero-client.ts

```diff
@@ -1,15 +1,120 @@
+
+  async getInfo(): Promise<GetInfoResultDto> {
+    return this.http
+      .post<{ result: GetInfoResultDto }>(
+        `${Config.blockchain.monero.d.url}/json_rpc`,
+        {
+          method: 'get_info',
+          params: { account_index: 0 },
+        },
+        this.httpConfig(),
+      )
+      .then((r) => r.result);
+  }
+
+  async getBlockHeight(): Promise<number> {
+    return this.http
+      .post<{ height: number }>(`${Config.blockchain.monero.d.url}/get_height`, {}, this.httpConfig())
+      .then((r) => r.height);
+  }
+
+  async getFeeEstimate(): Promise<GetFeeEstimateResultDto> {
+    return this.http
+      .post<{ result: GetFeeEstimateResultDto }>(
+        `${Config.blockchain.monero.d.url}/json_rpc`,
+        {
+          method: 'get_fee_estimate',
+          params: {},
+        },
+        this.httpConfig(),
+      )
+      .then((r) => this.convertFeeEstimateAuToXmr(r.result));
+  }
+
+  private convertFeeEstimateAuToXmr(feeEstimateResult: GetFeeEstimateResultDto): GetFeeEstimateResultDto {
+    feeEstimateResult.fee = MoneroHelper.auToXmr(feeEstimateResult.fee);
+    feeEstimateResult.fees = feeEstimateResult.fees.map((fee) => MoneroHelper.auToXmr(fee));
+
+    return feeEstimateResult;
+  }
+
+  async getTransaction(txId: string): Promise<MoneroTransactionDto | undefined> {
+    return this.http
+      .post<{ status: string; txs: GetTransactionResultDto[] }>(
+        `${Config.blockchain.monero.d.url}/get_transactions`,
+        {
+          txs_hashes: [txId],
+          decode_as_json: true,
+        },
+        this.httpConfig(),
+      )
+      .then((r) => this.mapTransaction(r.status, r.txs));
+  }
+
+  private mapTransaction(status: string, txs?: GetTransactionResultDto[]): MoneroTransactionDto {
+    if ('OK' !== status || !txs) return {};
+
+    const transactionResult = txs[0];
+
+    const txnAsJson = transactionResult.as_json;
+    const transaction = <MoneroTransactionDto>JSON.parse(txnAsJson);
+
+    transaction.block_height = transactionResult.block_height;
+    transaction.block_timestamp = transactionResult.block_timestamp;
+    transaction.confirmations = transactionResult.confirmations;
+    transaction.tx_hash = transactionResult.tx_hash;
+
+    transaction.txnFee = MoneroHelper.auToXmr(this.mapTransactionFee(transaction));
+
+    return transaction;
+  }
+
+  private mapTransactionFee(transaction: MoneroTransactionDto): number {
+    const vinAmounts = transaction.vin?.map((vin) => vin.key.amount) ?? [0];
+    const voutAmounts = transaction.vout?.map((vout) => vout.amount) ?? [0];
+
+    const totalVinAmount = Util.sum(vinAmounts);
+    const totalVoutAmount = Util.sum(voutAmounts);
+
+    transaction.inAmount = MoneroHelper.auToXmr(totalVinAmount);
+    transaction.outAmount = MoneroHelper.auToXmr(totalVoutAmount);
+
+    const transactionFee = totalVinAmount - totalVoutAmount;
+    if (transactionFee > 0) return transactionFee;
+
+    return transaction.rct_signatures?.txnFee ?? 0;

```

Comment:
Why is this?

Reply:
There are two different kinds of transactions in monero: Confidential and Non-Confidential.

Non-Confidential transactions has "vin" and "vout" data given and the fee has to be calculated from the "vin" and "vout" amounts.
The Confidential transactions are totally confidential and the fee is given within the "rct_signatures" data. No "vin" and "vout" amounts are available.

Example 1: d6e48158472848e6687173a91ae6eebfa3e1d778e65252ee99d7515d63090408
Example 2: 9929f4af096152e205971a334c55eb2e6fec85ab6905e1cba9a5e6246ea6dc33